dali/pipeline/operators/decoder/nvjpeg_decoder.cu

```diff
@@ -37,7 +39,22 @@ is encountered and internal buffer needs to be reallocated to decode it.)code",
       R"code(Padding for nvJPEG's host memory allocations in bytes.
 This parameter helps to avoid reallocation in nvJPEG whenever a bigger image
 is encountered and internal buffer needs to be reallocated to decode it.)code",
-      16*1024*1024);
+      16*1024*1024)
+  .AddOptionalArg("cache_size",
+      R"code(Total size of the decoder cache in bytes. When provided, decoded
+images bigger than `cache_threshold` will be cached in memory.)code",
+      0)
+  .AddOptionalArg("cache_threshold",
+      R"code(Size threshold (in bytes) for images to be cached.)code",
+      0)
+  .AddOptionalArg("cache_debug",
+      R"code(Print debug information about decoder cache.)code",
+      false)
+  .AddOptionalArg("cache_type",
+      R"code(Choose cache type:
+`threshold`: Caches every image with size bigger than `cache_threshold` until cache is full.
```

Comment:
Maybe some info about `stick_to_shard` loader option that should be used together.
Also please add some info about expected cache warm up time.

Reply:
added